src/core/backbone_router/local.cpp

```diff
@@ -238,6 +247,100 @@ void Local::UpdateBackboneRouterPrimary(Leader::State aState, const BackboneRout
     return;
 }
 
+otError Local::GetDomainPrefix(NetworkData::OnMeshPrefixConfig &aConfig)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(mDomainPrefixConfig.mPrefix.mLength > 0, error = OT_ERROR_NOT_FOUND);
+
+    aConfig = mDomainPrefixConfig;
+
+exit:
+    return error;
+}
+
+otError Local::RemoveDomainPrefix(const otIp6Prefix &aPrefix)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aPrefix.mLength > 0, error = OT_ERROR_INVALID_ARGS);
+
+    VerifyOrExit(mDomainPrefixConfig.mPrefix.mLength == aPrefix.mLength, error = OT_ERROR_NOT_FOUND);
+
+    VerifyOrExit(Ip6::Address::PrefixMatch(mDomainPrefixConfig.mPrefix.mPrefix.mFields.m8, aPrefix.mPrefix.mFields.m8,
+                                           BitVectorBytes(aPrefix.mLength)) >= aPrefix.mLength,
+                 error = OT_ERROR_NOT_FOUND);
+
+    if (IsEnabled())
+    {
+        RemoveDomainPrefixFromNetworkData();
+    }
+
+    mDomainPrefixConfig.mPrefix.mLength = 0;
+
+exit:
+    return error;
+}
+
+void Local::SetDomainPrefix(const NetworkData::OnMeshPrefixConfig &aConfig)
+{
+    if (IsEnabled())
+    {
+        RemoveDomainPrefixFromNetworkData();
+    }
+
+    mDomainPrefixConfig = aConfig;
+    LogDomainPrefix("Set", OT_ERROR_NONE);
+
+    if (IsEnabled())
+    {
+        AddDomainPrefixToNetworkData();
+    }
+}
+
+void Local::RemoveDomainPrefixFromNetworkData(void)
+{
+    otError error = OT_ERROR_NOT_FOUND; // only used for logging.
+
```

Comment:
```suggestion

    VerifyOrExit(IsEnabled(), error = OT_ERROR_INVALID_STATE); 
```
Since `RemoveDomainPrefixFromNetworkData` is always called when enabled, what about move checking `IsEnabled()` here to make calling `RemoveDomainPrefixFromNetworkData()` even simplier? No need to check have `if (IsEnabled()) { ... }`.

Reply:
also hesitated here... decided to do as is now with the considerations - 1) In `SetEnabled()`, it is determined for the Add or Remove action, 2) it might be easier to understand the logical when reading the code like `SetDomainPrefix()` .

https://github.com/openthread/openthread/pull/4793/files#diff-f0e917ba333de39b51d15d69eef03a3eR287

prefer a bit to keep it as is.

Reply:
Fine with keeping it as is for better readability.